crowdsourcing/viewsets/project.py

```diff
@@ -141,6 +142,81 @@ def requester_projects(self, request, **kwargs):
         serializer = ProjectSerializer(instance=projects, many=True, fields=('id', 'name', 'module_count'))
         return Response(serializer.data)
 
+    @list_route(methods=['GET'])
+    def workers_reviews(self, request, **kwargs):
+        projects = request.user.userprofile.requester.project_owner.all()
+        modules = Module.objects.all().filter(project__in=projects)
+        tasks = []
+        module_task_map = {}
+        for module in modules:
+          module_tasks = Task.objects.all().filter(module=module)
+          for tsk in module_tasks:
+            module_task_map[tsk.id] = tsk.module
+          tasks.extend(module_tasks)
+        task_workers = TaskWorker.objects.all().filter(
+            task__in=tasks, task_status__in=[2, 3, 4, 5])
+        serializer = TaskWorkerSerializer(instance=task_workers, many=True)
+        for entry in serializer.data:
+          entry["module"] = module_task_map[entry["task"]].id
+          entry["module_name"] = module_task_map[entry["task"]].name
+          entry["target"] = entry["worker"]
+
+        #dedupe by module and worker
+        pending_reviews = {}
+        for entry in serializer.data:
+          pending_reviews[(entry["module"], entry["worker"])] = entry
+
+        # Get existing ratings
+        ratings = WorkerRequesterRating.objects.all().filter(
+          origin=request.user.userprofile, module__in=modules, type="requester")
+        rating_map = {}
+        for rating in ratings:
+          rating_map[(rating.module.id, rating.target.id)] = rating
+
+        for key, val in rating_map.items():
+          if key in pending_reviews:
+            current_review = pending_reviews[key]
+            current_review["current_rating"] = val.weight
+            current_review["current_rating_id"] = val.id
```

Comment:
multiple db hits and loops will kill perfomance;
use something like the following which achieves what you need:
```
from django.db.models import F, Count
TaskWorker.objects.values('task__module__name', 'task__module__id', 
                          'worker__profile__rating_target__weight', 'worker__profile__id')\
    .filter(task__module__project__owner=request.user.userprofile.requester.id, task_status__in=[3,4,5]).extra(
    where=["(crowdsourcing_module.id=crowdsourcing_workerrequesterrating.module_id OR crowdsourcing_workerrequesterrating.module_id is null)"]
).annotate(task_count=Count('task'))
```